### DISCUSSION TITLE
Strange warning message in "Station Locations" view

### DISCUSSION THREAD

**Discussion by m0urs** [magicbug/Cloudlog#2094] [Q&A] [answered]
I see the following warning message in the station location view:

"Warning Due to recent changes within Cloudlog you need to reassign QSOs to your station profiles. Create a station profile, if you haven't already, then press this button to assign all QSOs to the first Station Profile."

I do have already several Station Profiles and I am unaware that I should have QSOs which do not have a station profile assigned.

What is the reason for this message and what happens if I press that button? I do not want to risk that all my QSOs are assigned to my main profile (although most of the QSOs should be asigned to that location profile already).

![image](https://user-images.githubusercontent.com/2588362/235064653-1552b545-bff4-4993-9125-3f28cbf866d4.png)

**Comment by phl0** [comment]
Which branch are you on? And did that message appear only after recent changes?

**Reply by m0urs** [reply]
I am on my own branch which is, however, almost 100% identical to the current DEV branch. I am not sure when this message appeared the first time, as I was not in that view for quite some time.

What exactly is the reason for that message? QSOs without a n assigned station location? If yes, what should I check in the DB to find these?

**Reply by m0urs** [reply]
These are, from my point of view, the relevant code line:

`application/views/station_profile/index.php:            <?php if($is_there_qsos_with_no_station_id >= 1) { ?>`

`application/controllers/Station.php:            $data['is_there_qsos_with_no_station_id'] = $this->Logbook_model->check_for_station_id();`

```
    public function check_for_station_id() {
      $this->db->where('station_id =', NULL);
      $query = $this->db->get($this->config->item('table_name'));
      if($query->num_rows() >= 1) {
        return 1;
      } else {
        return 0;
      }
    }

```
Am I right, that the function "check_for_station_id()" is testing, if there are QSO records with "station_id = NULL" and if there are such rows then it returns "1", if there are NO such records then it returns "0"?

In this case the IF condition "if($is_there_qsos_with_no_station_id >= 1)" would be TRUE if there are QSO records with "station_id = NULL2 and then it displays that warning.

However, I do not seem to have sich records?

![image](https://user-images.githubusercontent.com/2588362/235074025-7209a6fb-f446-48e1-991d-af3dd8d2b46e.png)

So the message should not appear?

**Reply by phl0** [reply]
This is indeed strange. Just for the case could you check if there are data sets where `station_id` is 0 (i.e. not NULL)?

**Reply by m0urs** [reply]
Yes, I tested that already before:
![image](https://user-images.githubusercontent.com/2588362/235184298-43a03c07-56f5-408f-beb4-5ff0936c6b8e.png)

**Reply by m0urs** [reply]
You need to do the SQL reuest correctly, then you do get a correct result. Sorry, my error.

![image](https://user-images.githubusercontent.com/2588362/235194888-4243134d-9b3a-4a78-b971-8abda0860993.png)

**Reply by m0urs** [reply, marked as answer]
All these records have either all fields set to NULL or do have QSOs which are already correctly logged. So it seems that they are somehow duplicates. They are all from the same two days in end of June 2022. I guess something was wrong on these days (maybe I did run a test version of Cloudlog at that time). I checked all records and all of them were duplicates to existing QSOs which had the station_id correctly assigned. So I deleted all these records and the message is gone now :-) I am quite sure that I was inside that view after 2022-06-30 ... So I have no idea when these duplicate entries have been created ... 

But the issue is solved now .... Thanks !
